### PR TITLE
Feat/hasura jwt auth

### PR DESCRIPTION
**Pull request by kwiss**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: "3"
+version: '3'
```
Comment (douglasduteil) [inline docker-compose.yaml]: Should be double quotes. Can you prettify that file ?

File: docker-compose.yaml [hunk 2]
```diff
@@ -26,8 +35,30 @@ services:
...
+  graphql-engine:
+    image: hasura/graphql-engine:v1.0.0-beta.2
+    ports:
+      - 5000:8080
+    depends_on:
+      - db
+    networks:
+      default:
+      outside:
+        ipv4_address: 192.168.220.220
```
Comment (douglasduteil) [inline docker-compose.yaml]: why aren't you using the default network ?

File: packages/api/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+ * Sign in using username and password and returns JWT
+ */
+exports.postLogin = async (req, res, next) => {
+  passport.authenticate("local", (err, user, info) => {
+    if (err) {
+      return next(err);
+    }
+    if (!user) {
+      if (process.env.NODE_ENV !== "test") {
+        console.log(`Unauthorized user : ${req.body.username}`);
```
Comment (douglasduteil) [inline packages/api/controllers/user.js]: Is it necessary ?

File: packages/api/controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        .json({ success: false, message: "User not found" });
+    }
+    if (user) {
+      req.logIn(user, { session: false }, function(err) {
+        if (err) {
+          return next(err);
+        }
+
+        updateLastLogin(user.id)
+          .then(() =>
```
Comment (douglasduteil) [inline packages/api/controllers/user.js]: Prefer using async await notation

File: packages/api/controllers/user.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+              token: getJwt(user.id, user.username)
+            });
+          })
+          .catch(e => {
+            addDataLogs({
+              user_id: user.id,
+              action: "connexion",
+              result: "fail"
+            });
+            return console.log(e);
```
Comment (douglasduteil) [inline packages/api/controllers/user.js]: should be a `next(e)` or `throw e` instead no ?

File: packages/api/controllers/user.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    }
+    if (user) {
+      handleResponse(res, 200, user.getHasuraClaims());
+    } else {
+      handleResponse(res, 200, { "X-Hasura-Role": "anonymous" });
+    }
+  })(req, res, next);
+};
+
+function handleResponse(res, code, statusMsg) {
```
Comment (douglasduteil) [inline packages/api/controllers/user.js]: Is this one line wrapper necessary ?

File: packages/api/controllers/user.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+      });
+    }
+  })(req, res, next);
+};
+
+// Webhook can be used with hasura
+exports.getWebhook = async (req, res, next) => {
+  passport.authenticate("bearer", (err, user, info) => {
+    if (err) {
+      return handleResponse(res, 401, { error: err });
```
Comment (douglasduteil) [inline packages/api/controllers/user.js]: Prefer https://www.npmjs.com/package/http-errors

File: packages/api/package.json [hunk 8]
```diff
@@ -35,16 +35,19 @@
...
     "pg": "^4.5.3",
     "pg-native": "^3.0.0",
     "query-string": "^6.5.0",
     "rand-token": "^0.4.0",
+    "rasha": "^1.2.5",
     "serve-favicon": "^2.5.0",
     "swagger-jsdoc": "^3.2.9",
-    "swagger-ui-dist": "^3.22.1"
+    "swagger-ui-dist": "^3.22.1",
+    "util": "^0.12.0"
```
Comment (douglasduteil) [inline packages/api/package.json]: Why do we need `util` here ?

File: packages/api/private.pem [hunk 9]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (douglasduteil) [inline packages/api/private.pem]: Should it be in the versioned public source code ?

File: packages/api/public.pem [hunk 10]
```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
```
Comment (douglasduteil) [inline packages/api/public.pem]: Should it be in the versioned public source code ?

File: packages/api/private.pem (hunk 9 above)
Comment (kwiss) [reply packages/api/private.pem]: no it's an example file, it need to be remove before merge and production 
=> // WARNING: .pem files are here for exemple don't user them in production in 
https://github.com/SocialGouv/emjpm/blob/fe9e92697a1d494386f7db047c0e2af8b56fc6b2/packages/api/config/jwt.js#L5

File: packages/api/package.json (hunk 8 above)
Comment (kwiss) [reply packages/api/package.json]: need to be removed

**Pull request closed without merging**